main.go

```diff
@@ -73,6 +99,17 @@ func checkFiles(w io.Writer, rootRule rules.Rule, cfg *config.Config, files []so
 		// `allowUnmatched` is explicitly set to `true` in the configuration.
 		if !result.Success || (result.Unmatched.Size() > 0 && !cfg.AllowUnmatched) {
 			problematicFiles = append(problematicFiles, file.Path)
+
+			sarifResult := result.SarifResult(file.Path)
+			results = append(results, sarifResult)
```

Comment:
I just realized that this only produces results for failures. A sarif report also contains the results of successful checks, so we need to move this out of the `if`.

We should also add a test case for the happy path here.